instrumentation/oshi/library/build.gradle.kts

```diff
@@ -3,11 +3,13 @@ plugins {
   id("com.google.osdetector")
 }
 
-// 5.5.0 is the first version that works on arm mac
-val oshiVersion = if (osdetector.os == "osx" && osdetector.arch == "aarch_64") "5.5.0" else "5.3.1"
-
 dependencies {
-  library("com.github.oshi:oshi-core:$oshiVersion")
+  library("com.github.oshi:oshi-core:5.3.1")
+
+  if (osdetector.os == "osx" && osdetector.arch == "aarch_64" && !(findProperty("testLatestDeps") as Boolean)) {
+    // 5.5.0 is the first version that works on arm mac
+    configurations.testRuntimeClasspath.get().resolutionStrategy.force("com.github.oshi:oshi-core:5.5.0")
+  }
```

Comment:
usually we have stuff like this outside of the dependencies block, like in https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/826ba7402e73d36e8c683ea654ca709bd1809e65/instrumentation/jaxrs/jaxrs-2.0/jaxrs-2.0-jersey-2.0/javaagent/build.gradle.kts#L53